x-pack/plugins/apm/public/components/routing/templates/apm_service_template/analyze_data_button.tsx

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EuiButtonEmpty, EuiToolTip } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import React from 'react';
+import { useKibana } from '../../../../../../../../src/plugins/kibana_react/public';
+import {
+  createExploratoryViewUrl,
+  SeriesUrl,
+} from '../../../../../../observability/public';
+import {
+  isIosAgentName,
+  isRumAgentName,
+} from '../../../../../common/agent_name';
+import {
+  SERVICE_ENVIRONMENT,
+  SERVICE_NAME,
+} from '../../../../../common/elasticsearch_fieldnames';
+import {
+  ENVIRONMENT_ALL,
+  ENVIRONMENT_NOT_DEFINED,
+} from '../../../../../common/environment_filter_values';
+import { useApmServiceContext } from '../../../../context/apm_service/use_apm_service_context';
+import { useUrlParams } from '../../../../context/url_params_context/use_url_params';
+
+interface AnalyzeDataButtonProps {
+  serviceName: string;
+}
+
+export function AnalyzeDataButton({ serviceName }: AnalyzeDataButtonProps) {
+  const { agentName } = useApmServiceContext();
+  const { services } = useKibana();
+  const { urlParams } = useUrlParams();
+  const { rangeTo, rangeFrom, environment } = urlParams;
+  const basepath = services.http?.basePath.get();
+
+  const environmentDefinition =
+    !environment ||
+    environment === ENVIRONMENT_NOT_DEFINED.value ||
+    environment === ENVIRONMENT_ALL.value
+      ? {}
```

Comment:
this will also work, but if you explicitly wants to select ALL_Values in some field in exploratory view, you can use 'ALL_VALUES' as field, there is a constant declared 

it will become 

`[SERVICE_ENVIRONMENT]: [ALL_VALUES_SELECTED]`

https://github.com/elastic/kibana/blob/master/x-pack/plugins/observability/public/components/shared/field_value_suggestions/field_value_combobox.tsx#L14